dialogue-apache-hc5-client/src/test/java/com/palantir/dialogue/hc5/NoResponseTest.java

```diff
@@ -76,8 +77,8 @@ public void testConnectionClosedAfterDelay() {
         }));
         try {
             Channel channel = create(defaultClientConfig(getPort(server)));
-            assertThatThrownBy(channel.execute(TestEndpoint.POST, request)::get)
-                    .hasCauseInstanceOf(SocketTimeoutException.class);
+            ListenableFuture<Response> response = channel.execute(TestEndpoint.POST, request);
+            assertThatThrownBy(() -> response.get()).hasCauseInstanceOf(SocketTimeoutException.class);
```

Comment:
This assertj upgrade adds an overload to `assertThatThrownBy` which breaks some method reference inputs. We may want to excavate something to handle this.

In this example I extracted a variable, but it may be easier on the automation if we add a cast to the specific assertj interface type, otherwise the `assertThatThrownBy` may accept exceptions thrown from a broader snippet than we expect. For example, we want this test to fail if `channel.execute` throws a SocketTimeoutException, which should only be thrown by `response.get()`.

Reply:
Disregard, this has been reverted in assertj: https://github.com/assertj/assertj/releases/tag/assertj-build-3.25.1